packages/aws-cdk-lib/aws-cloudfront-origins/test/s3-static-website-origin.test.ts

```diff
@@ -0,0 +1,70 @@
+import * as cloudfront from '../../aws-cloudfront';
+import * as origins from '../../aws-cloudfront-origins';
+import * as s3 from '../../aws-s3';
+import { Duration, Stack } from '../../core';
+
+describe('S3StaticWebsiteOrigin', () => {
+  let stack: Stack;
+  let bucket: s3.Bucket;
+
+  beforeEach(() => {
+    stack = new Stack();
+    bucket = new s3.Bucket(stack, 'myStaticWebsiteBucket', {
+      websiteIndexDocument: 'index.html',
+    });
+  });
+  test('creates a S3 static website origin with default properties set', () => {
+    const origin = new origins.S3StaticWebsiteOrigin({ bucket });
+    const originBindConfig = origin.bind(stack, { originId: 'StackOrigin12AB34569' });
+
+    expect(originBindConfig.originProperty).toEqual({
+      id: 'StackOrigin12AB34569',
+      domainName: bucket.bucketWebsiteDomainName,
+      customOriginConfig: {
+        originSslProtocols: ['TLSv1.2'],
```

Comment:
If S3 website endpoint support HTTP only and not HTTPS, why do we need this `originSslProtocols` and what would it do?

Reply:
Good point. We don't need it but it's currently set by default in the constructor for `HttpOrigin` [here](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-cloudfront-origins/lib/http-origin.ts#L74). It does not have any effect on the deployment though and the origin protocol policy is correctly set as HTTP only.